server/src/main/java/org/apache/iotdb/db/utils/timerangeiterator/AggrWindowIterator.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iotdb.db.utils.timerangeiterator;
+
+import org.apache.iotdb.db.qp.utils.DatetimeUtils;
+import org.apache.iotdb.tsfile.utils.Pair;
+
+public class AggrWindowIterator implements ITimeRangeIterator {
+
+  // total query [startTime, endTime)
+  private final long startTime;
+  private final long endTime;
+
+  private final long interval;
+  private final long slidingStep;
+
+  private final boolean isAscending;
+  private final boolean isSlidingStepByMonth;
+  private final boolean isIntervalByMonth;
+
+  private static final long MS_TO_MONTH = 30 * 86400_000L;
```

Comment:
This constant is decleared in more than 2 classes, maybe we can move it to some class storing constants?

Reply:
fixed